packages/crypto/src/crypto.ts

```diff
@@ -466,3 +493,183 @@ const bigIntToHex = (num: bigint, length: number): string => {
+    );
+  }
+  index++; // Move past the INTEGER tag
+  const rLength = derSignatureBuf[index];
+  index++; // Move past the length byte
+  const r = derSignatureBuf.slice(index, index + rLength!);
+  index += rLength!; // Move to the start of s
+
+  // Parse 's' and check for integer tag (0x02)
+  if (derSignatureBuf[index] !== 0x02) {
+    throw new Error(
+      "failed to convert DER-encoded signature: invalid tag for s"
+    );
+  }
+  index++; // Move past the INTEGER tag
+  const sLength = derSignatureBuf[index];
+  index++; // Move past the length byte
+  const s = derSignatureBuf.slice(index, index + sLength!);
+
+  // Normalize 'r' and 's' to 32 bytes each
+  const rPadded = normalizePadding(r, 32);
+  const sPadded = normalizePadding(s, 32);
+
+  // Concatenate and return the raw signature
+  return new Uint8Array([...rPadded, ...sPadded]);
+};
+
+/**
+ * Function to normalize padding of byte array with 0's to a target length.
+ *
+ * @param {Uint8Array} byteArray - The byte array to pad or trim.
+ * @param {number} targetLength - The target length after padding or trimming.
+ * @returns {Uint8Array} - The normalized byte array.
+ */
+const normalizePadding = (byteArray: Uint8Array, targetLength: number) => {
+  const paddingLength = targetLength - byteArray.length;
+
+  // Add leading 0's to array
+  if (paddingLength > 0) {
+    const padding = new Uint8Array(paddingLength).fill(0);
+    return new Uint8Array([...padding, ...byteArray]);
+  }
+
+  // Remove leading 0's from array
+  if (paddingLength < 0) {
+    const expectedZeroCount = paddingLength * -1;
+    let zeroCount = 0;
+    for (let i = 0; i < expectedZeroCount && i < byteArray.length; i++) {
+      if (byteArray[i] === 0) {
+        zeroCount++;
+      }
+    }
+    // Check if the number of zeros found equals the number of zeroes expected
+    if (zeroCount !== expectedZeroCount) {
+      throw new Error(
+        `invalid number of starting zeroes. Expected number of zeroes: ${expectedZeroCount}. Found: ${zeroCount}.`
+      );
+    }
+    return byteArray.slice(expectedZeroCount, expectedZeroCount + targetLength);
+  }
+  return byteArray;
+};
+
+/**
+ * Loads an ECDSA public key from a raw format for signature verification.
+ *
+ * @param {Uint8Array} quorumPublic - The raw public key bytes.
+ * @returns {Promise<CryptoKey>} - The imported ECDSA public key.
+ */
+const loadQuorumKey = async (quorumPublic: Uint8Array) => {
+  return await crypto.subtle.importKey(
+    "raw",
+    quorumPublic,
+    {
+      name: "ECDSA",
+      namedCurve: "P-256",
+    },
+    true,
+    ["verify"]
+  );
+};
+
+/**
+ * Decodes a private key based on the specified format.
+ *
+ * @param {string} privateKey - The private key to decode.
+ * @param {string} keyFormat - The format of the private key (e.g., "SOLANA", "HEXADECIMAL").
+ * @returns {Uint8Array} - The decoded private key.
+ */
+
+export const decodeKey = (privateKey: string, keyFormat: any) => {
```

Comment:
let's add the return type to the function signature
```suggestion
export const decodeKey = (privateKey: string, keyFormat: any) : Uint8Array => {
```